terraform/transform_provider_test.go

```diff
@@ -785,3 +581,12 @@ module.child.module.grandchild.aws_instance.baz
   provider["registry.terraform.io/-/aws"].foo
 provider["registry.terraform.io/-/aws"].foo
 `
+
+const testTransformImportModuleChildStr = `        
+module.child.aws_instance.foo
+  provider["registry.terraform.io/-/aws"]
+module.child.aws_instance.foo (import id "bar")
+  provider["registry.terraform.io/-/aws"]
+module.child.module.nested.aws_instance.foo
```

Comment:
Where is this nested instance coming from? I don't see it in the test above

Reply:
Ah, I see it now, it's in the new config. Should this be showing up here, since we're not importing it?

Reply:
I got the impression that it's expected to show the entire state, not just what was imported. Does that match your expectation? (the previous version of this test was importing into an empty config).

Reply:
This is the graph, and not the state however. I would do some tests to see what happens when you import 1 resource, but there are other resources in the config (both existent and non-existent). This might be fine and we can fix it later, but I'm afraid that this might error out once the import process starts.

Reply:
Smoke test w/a config with two random_ids, one existing in state and the other not:

```
$terraform import random_id.dos TdTfsRX6g4o
random_id.dos: Importing from ID "TdTfsRX6g4o"...
random_id.dos: Import prepared!
  Prepared random_id for import
random_id.dos: Refreshing state... [id=TdTfsRX6g4o]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

$ terraform import random_id.nonexist TdTfsRX6g4o
Error: resource address "random_id.nonexist" does not exist in the configuration.

Before importing this resource, please create its configuration in the root module. For example:

resource "random_id" "nonexist" {
  # (resource arguments)
}```